spec/models/zip_code_spec.rb

```diff
@@ -1,7 +1,72 @@
 # frozen_string_literal: true
 
 require 'rails_helper'
+require './spec/support/test_helper'
 
 RSpec.describe ZipCode, type: :model do
-  it { should validate_presence_of(:d_codigo) }
+  include Helper
+
+  context 'validations tests' do
+    it { should validate_presence_of(:d_codigo) }
+  end
+
+  context 'scopes tests' do
+    let!(:zip_code) { FactoryBot.create(:zip_code) }
+
+    it 'should return a zip_code' do
+      expect(ZipCode.count).to eq(1)
+    end
+
+    it 'search for a wrong zip_code gives empty results' do
+      create_cp(%w[57300 21360 81920])
+      cp = '64000'
+
+      expect(ZipCode.find_by_zip_code(cp)).to be_empty
+    end
+
+    it 'search for a valid zip_code gives valid results' do
+      cp = '01000'
+
+      expect(ZipCode.find_by_zip_code(cp)).to eq([ZipCode.last])
+    end
+
+    it 'search for a wrong state gives empty results' do
+      create_states(%w[Tamaulipas Sinaloa Tijuana])
+      state = 'Nuevo Leon'
+
+      expect(ZipCode.find_by_state(state)).to be_empty
+    end
+
+    it 'search for a valid state gives valid results' do
+      state = 'Ciudad de México'
+
+      expect(ZipCode.find_by_state(state)).to eq([ZipCode.last])
```

Comment:
Here is not clear why it should be `eq` to the last ZipCode, please be more explicit on this.

If for some reason you create another ZipCode, would this work?

Reply:
Done!